src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter_IsPatternOperator.cs

```diff
@@ -109,8 +109,15 @@ internal BoundExpression LowerGeneralIsPattern(BoundIsPatternExpression node)
                 _factory.Syntax = node.Syntax;
                 var resultBuilder = ArrayBuilder<BoundStatement>.GetInstance();
                 var inputExpression = _localRewriter.VisitExpression(node.Expression);
-                BoundDecisionDag decisionDag = ShareTempsIfPossibleAndEvaluateInput(
-                    node.DecisionDag, inputExpression, resultBuilder, out _);
+
+                BoundDecisionDag decisionDag = node.DecisionDag;
```

Comment:
>DecisionDag [](http://example.com/codeflow?start=52&length=11)

It might be useful to rename properties like this in bound nodes to something like "DecisionDagNotForCodeGeneration"

Reply:
Only a single decision dag is kept in the bound tree. The other is created on the fly if needed.
(Plus DecisionDagNotForCodeGeneration is not true. Normally, we *do* use it for codegen.)

Reply:
> Only a single decision dag is kept in the bound tree. The other is created on the fly if needed.
(Plus DecisionDagNotForCodeGeneration is not true. Normally, we do use it for codegen.)

The goal is to prevent people from using this Dag for codegen without special considerations first. It does not matter much if we end up reusing the Dag at the end, the main part is that we are supposed to check if it is reusable first. An "alarm" in the name helps to force a developer (and code reviewers as well) to think about the usage from that perspective. Feel free to use a "better" name.

Reply:
> How do you feel about an instance method GetDecisionDagForLowering that does this check? We can also rename the field to ReachabilityDecisionDag for explicit signal on plain usages.

I think that will address my concern. There is still a question in what type ```GetDecisionDagForLowering``` will be defined, but having a dedicated helper feels like a good idea.